### PR TITLE
139: Quote Block with Avatar, Link, and Background color variations

### LINKED ISSUE #139: L2 Quote Block
Follow-up to L3 quote block https://github.com/urfuwo/hlx-test/issues/107

L2 quotes can have an additional avatar and a link (while L3 quotes could only have a quote text and a quote source)

Design Specs:
* Figma L2 quote block: https://www.figma.com/file/oSetT4LbatRmXlcB2A7V8Z/Content-Hubs-2024?type=design&node-id=1507-65208&mode=design&t=pcs6BzMbBgLQPBw2-4
* Figma Component Spec: https://www.figma.com/file/OFEfvhyXJvYJ6MQZ30699m/v2.0-Brand-Digital-Design-System?type=design&node-id=6526-15728&mode=design&t=Aq7vEK0dS8OPWFrD-0
* Frontify Spec: https://sap.frontify.com/document/223142#/-/quote

### PR DESCRIPTION
**Pull request by rjwtrmn**
- Added avatar (left and right variant as per [Frontify](https://sap.frontify.com/document/223142#/-/quote)) on L2 pages only
- Added Link on L2 pages only
- Background colour achieved with section wrapper

Fix #139

Test URLs:

L2
- Before: https://main--hlx-test--urfuwo.hlx.live/draft/rjwtrmn/thomassaueressig-copy
- After: https://139-quote-block-variations--hlx-test--urfuwo.hlx.live/draft/rjwtrmn/thomassaueressig-copy

L3
- Before: https://main--hlx-test--urfuwo.hlx.live/draft/rjwtrmn/the-article-copy
- After: https://139-quote-block-variations--hlx-test--urfuwo.hlx.live/draft/rjwtrmn/the-article-copy

### CODE REVIEW

File: aemedge/libs/avatar/avatar.js [hunk 1]
```diff
@@ -17,19 +17,24 @@ export default class Avatar {
...
-      div({ class: `avatar ${size}` }, this.image ? div(this.getOptimizedPicture()) : div()),
-      div(
-        { class: 'avatar-info' },
-        div({ class: 'title' }, a({ href: this.path }, div(`${this.title}`))),
-        this.description ? div({ class: 'description info' }, this.description) : '',
-      ),
-    );
+    let element;
+    if (imageOnly) {
+      element = div({ class: `avatar ${size}` }, this.image ? div(this.getOptimizedPicture()) : div());
```
Comment (karlpauls) [inline aemedge/libs/avatar/avatar.js:26]: I think here you could return early - saves you the nesting.

File: aemedge/blocks/quote/quote.js [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+  const isSmall = block.classList.contains('small');
   qtContent.classList.add('col', 'content');
   qtContent.parentNode.classList.add('qt');
   if (qsContent) {
     qsContent.classList.add('col', 'content');
     qsContent.parentNode.classList.add('qs');
-  } else {
+    if (isNotArticle) {
+      const author = removeAuthorsSuffixes(qsContent.textContent).split(',')[0].trim();
+      const authorEntry = (await getAuthorEntries([author]))[0];
```
Comment (karlpauls) [inline aemedge/blocks/quote/quote.js:18]: Maybe use destructuring.

File: aemedge/blocks/quote/quote.css [hunk 3]
```diff
@@ -64,10 +106,10 @@
     padding-left: 16px;
 }
 
+.quote .quote-link,
 .quote .qs .content {
-    font-size: var(--udexTypographyHeadingRegularXXSFontSize);
-    padding-left: 46px;
-    padding-right: 46px;
+    margin-left: 46px;
```
Comment (karlpauls) [inline aemedge/blocks/quote/quote.css:111]: should those margins be variables?

File: aemedge/blocks/quote/quote.css [hunk 4]
```diff
@@ -90,8 +132,43 @@
     padding-left: 8px;
 }
 
+.quote.small .quote-link,
+.quote.small .qs .content {
+    margin-left: 36px;
```
Comment (karlpauls) [inline aemedge/blocks/quote/quote.css:137]: Same question for variables here.

File: aemedge/blocks/quote/quote.js [hunk 5]
```diff
@@ -1,12 +1,39 @@
...
+      if (authorEntry) {
+        const avatar = new Avatar(authorEntry.title, null, authorEntry.path, authorEntry.image);
+        block.insertBefore(avatar.render(isSmall ? 'medium' : 'big', false, true), qtContent.parentNode);
+      }
+    }
+  }
+  if (linkContent && isNotArticle) {
+    // Unwrap link from button container added by aem.js
+    const link = linkContent.querySelector('a');
+    link.classList = [];
```
Comment (karlpauls) [inline aemedge/blocks/quote/quote.js:28]: I think you could populate the list directly with the values from the next line, no?

**Review by karlpauls — APPROVED** [review]

File: aemedge/blocks/quote/quote.css (hunk 3 above)
Comment (rjwtrmn) [reply aemedge/blocks/quote/quote.css:111]: Sadly no variable exists for this

File: aemedge/blocks/quote/quote.css (hunk 4 above)
Comment (rjwtrmn) [reply aemedge/blocks/quote/quote.css:137]: Same here, no variable for 36px

**Review by mhaack — APPROVED** [review]

**Comment by saurabh-khare** [comment]
@rjwtrmn Please merge the PR

**Pull request merged**